### DISCUSSION TITLE
Pull out customized frequency from trading-IG historical data

### DISCUSSION THREAD

**Discussion by kiann00** [ig-python/ig-markets-api-python-library#279] [Q&A] [answered]
Hi there, for this function ig_service.fetch_historical_prices_by_epic_and_num_points()

My understanding is that, we specify the resolution  = day, minute, while num_points = number of points to retrieve.
I (just) need the 5min-time-frame. Is there a way to specify it directly, so as to limit my data-call quota?

I don't really want to download a huge set of 1-min timeframe data, and then use pandas, as_freq /resample to resample it to a 5min time-frame........

Thanks everyone! Appreciate your help, in your busy time

**Comment by starsnpixel** [comment, marked as answer]
I'm not sure if I understand your question correctly. If you check out the function definition in rest.py, you'll find
```
:param resolution: (str, optional) timescale resolution. Expected values
            are 1Min, 2Min, 3Min, 5Min, 10Min, 15Min, 30Min, 1H, 2H, 3H, 4H, D,
            W, M. Default is 1Min
```
So if you only want the 5min timeframe, just call it by `ig_service.fetch_historical_prices_by_epic_and_num_points(epic, '5Min', ...)` it should do exactly what you're looking for.

**Reply by kiann00** [reply]
@starsnpixel , many thanks. I missed the reading it in rest.py.... Thanks so much.

Follow-up confirmation, ..... since (market)data limit in IG is very tight. I believe we get like 10,000 pts per week.
So, if I use the 5min-timeframe for this, I presume I will 'eat' alot less data (i.e. the frame does not take in 1min time-frame and re-samples it to 5min)

**Reply by oliverpolden** [reply]
Yes you will use a lot of data at 5min. I personally used the streaming api for 5 min data. Checking my code I have the following to logic for a new candle and then write the old one to the database:
```        
        for currency in self.currencies:
            if currency in item_update["name"]:
                utm = item_update["values"]["UTM"]
                if self.candles[currency] and (utm > self.candles[currency]["UTM"]):
                    # Save candle to database.
```
item_update is in the example code and the above logic goes in my on_prices_update(item_update) function.
https://github.com/ig-python/trading-ig/blob/master/sample/stream_ig.py

**Reply by kiann00** [reply]
thanks @oliverpolden , .... yeah, using up my data-limit in the API is what I am afraid of.

So, I did investigate using the streaming as you advised here : https://github.com/ig-python/trading-ig/blob/master/sample/stream_ig.py

However, for some reason, I got an error trying to connect.... I posted my error message query there, where I put in my full code...

I don't suppose you know what is wrong? https://github.com/ig-python/trading-ig/discussions/281

**Reply by starsnpixel** [reply]
Hi there, 
thanks a lot for your example. I'm currently struggling with lightstreamer, thus this helps a lot. However, I've got two questions:
1) What does utm do in `utm = item_update["values"]["UTM"]`?
2) What's `self.candles`? Is a function your wrote yourself?

> Yes you will use a lot of data at 5min. I personally used the streaming api for 5 min data. Checking my code I have the following to logic for a new candle and then write the old one to the database:
> 
> ```
>         for currency in self.currencies:
>             if currency in item_update["name"]:
>                 utm = item_update["values"]["UTM"]
>                 if self.candles[currency] and (utm > self.candles[currency]["UTM"]):
>                     # Save candle to database.
> ```
> 
> item_update is in the example code and the above logic goes in my on_prices_update(item_update) function. https://github.com/ig-python/trading-ig/blob/master/sample/stream_ig.py

**Reply by starsnpixel** [reply]
> Yes you will use a lot of data at 5min. I personally used the streaming api for 5 min data. Checking my code I have the following to logic for a new candle and then write the old one to the database:
> 
> ```
>         for currency in self.currencies:
>             if currency in item_update["name"]:
>                 utm = item_update["values"]["UTM"]
>                 if self.candles[currency] and (utm > self.candles[currency]["UTM"]):
>                     # Save candle to database.
> ```
> 
> item_update is in the example code and the above logic goes in my on_prices_update(item_update) function. https://github.com/ig-python/trading-ig/blob/master/sample/stream_ig.py

Okay, I just looked deeper into it and found out that my questions weren't really clever. :D But could you please explain the logic behind `if self.candles[currency] and (utm > self.candles[currency]["UTM"])`?
